govcd/system_test.go

```diff
@@ -757,3 +757,83 @@ func (vcd *TestVCD) TestQueryAllVdcs(check *C) {
 		check.Assert(contains(knownVdcName, foundVdcNames), Equals, true)
 	}
 }
+
+// Test retrieval of all Orgs
+func (vcd *TestVCD) Test_QueryAllOrgs(check *C) {
+	vcd.skipIfNotSysAdmin(check)
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_QueryOrgByName: Org Name not given")
+		return
+	}
+
+	orgs, err := vcd.client.QueryAllOrgs()
+	check.Assert(err, IsNil)
+	check.Assert(orgs, NotNil)
+
+	if vcd.config.VCD.Org != "" {
+		foundOrg := false
+		for _, org := range orgs {
+			if org.Name == vcd.config.VCD.Org {
+				foundOrg = true
+			}
+		}
+		check.Assert(foundOrg, Equals, true)
+	}
+}
+
+// Tests Org retrieval by name, by ID, and by a combination of name and ID
+func (vcd *TestVCD) Test_QueryOrgByName(check *C) {
+	vcd.skipIfNotSysAdmin(check)
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_QueryOrgByName: Org Name not given")
+		return
+	}
+
+	org, err := vcd.client.QueryOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+
+	orgFound := false
+	if vcd.config.VCD.Org == org.Name {
+		orgFound = true
+	}
+
+	if testVerbose {
+		fmt.Printf("Org %s\n", org.Name)
+		fmt.Printf("\t href    %s\n", org.HREF)
+		fmt.Printf("\t enabled %v\n", org.IsEnabled)
+		fmt.Println("")
+	}
+
+	check.Assert(orgFound, Equals, true)
+}
+
+// Tests Org retrieval by name, by ID, and by a combination of name and ID
+func (vcd *TestVCD) Test_QueryOrgById(check *C) {
+	vcd.skipIfNotSysAdmin(check)
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_QueryOrgByName: Org Name not given")
+		return
+	}
+
+	namedOrg, err := vcd.client.QueryOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+
+	orgFound := false
+	if vcd.config.VCD.Org == namedOrg.Name {
+
+		idOrg, err := vcd.client.QueryOrgByID(namedOrg.ID)
+		check.Assert(err, IsNil)
+
+		if idOrg.ID == namedOrg.ID {
+			orgFound = true
+		}
+
+		if testVerbose {
+			fmt.Printf("Org %s\n", namedOrg.Name)
+			fmt.Printf("\t Org ID (by Name): %s\n", namedOrg.ID)
+			fmt.Printf("\t Org ID (by ID): %s\n", idOrg.ID)
+			fmt.Println("")
```

Comment:
The ID is not returned by the query system. Better using the HREF.

Reply:
I can reproduce the missing ID. Is that a VCD bug? According to the [API](https://developer.vmware.com/apis/1703/vmware-cloud-director/doc/doc//types/QueryResultOrgRecordType.html) the OrgRecord type is supposed to return an id field (since v1.5 up and including v10.5).

I ammended the test to use HREF in the next commit.